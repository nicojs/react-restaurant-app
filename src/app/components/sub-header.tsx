import { PropsWithChildren } from 'react';

export function SubHeader({ children }: PropsWithChildren) {
  return <h2 className="text-xl my-3 font-bold">{children}</h2>;
}
