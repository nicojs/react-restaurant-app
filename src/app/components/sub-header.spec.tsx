import { render, screen  } from '@testing-library/react';
import { SubHeader } from './sub-header';

describe(SubHeader, () => {
  it('should render an h2', () => {
    const thing = render(<SubHeader>Hello world</SubHeader>);
    const h2 = thing.getByText('Hello world');
    expect(h2.tagName).toBe('H2');
    expect(h2.className).toBe('text-xl my-3 font-bold');
  });

  it('should match snapshot', () => {
    const fragment = render(<SubHeader>Hello world</SubHeader>).asFragment();
    expect(fragment).toMatchSnapshot();
  });
});
