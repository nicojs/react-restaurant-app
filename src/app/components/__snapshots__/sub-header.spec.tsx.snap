// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader should match snapshot 1`] = `
<DocumentFragment>
  <h2
    class="text-xl my-3 font-bold"
  >
    Hello world
  </h2>
</DocumentFragment>
`;
